src/components/wasteCalendar/WasteReminderSettings.tsx

```diff
@@ -0,0 +1,361 @@
+import React, { useCallback, useEffect, useReducer, useState } from 'react';
+import {
+  Alert,
+  FlatList,
+  Modal,
+  RefreshControl,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+  TouchableOpacity,
+  View
+} from 'react-native';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { ListItem, normalize } from 'react-native-elements';
+
+import {
+  deleteReminderSetting,
+  getReminderSettings,
+  updateReminderSettings
+} from '../../pushNotifications';
+import { Switch } from '../Switch';
+import { BoldText, RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Radiobutton } from '../Radiobutton';
+import { Wrapper, WrapperHorizontal, WrapperRow } from '../Wrapper';
+import { colors, device, texts } from '../../config';
+import { ReminderSettings, WasteTypeData } from '../../types';
+import {
+  ReminderSettingsAction,
+  ReminderSettingsActionType,
+  reminderSettingsReducer
+} from './ReminderSettingsReducer';
+import { Button } from '../Button';
+import { areValidReminderSettings, parseReminderSettings } from '../../jsonValidation';
+import { SectionHeader } from '../SectionHeader';
+
+const showErrorAlert = () => {
+  Alert.alert(texts.wasteCalendar.errorOnUpdateTitle, texts.wasteCalendar.errorOnUpdateBody);
+};
+
+const initialReminderTime = new Date();
+initialReminderTime.setHours(9);
+initialReminderTime.setMinutes(0);
+
+const initialSettings: ReminderSettings = {
+  activeTypes: {},
+  onDayBefore: true,
+  reminderTime: initialReminderTime
+};
+
+const formatTime = (time: Date) => {
+  const minutes = time.getMinutes();
+  const hours = time.getHours();
+
+  let minutesString = minutes.toString();
+  if (minutes < 10) minutesString = `0${minutesString}`;
+
+  return `${hours}:${minutesString} Uhr`;
+};
+
+const keyExtractor = (item: string) => item;
+
+const CategoryEntry = ({
+  active,
+  categoryKey,
+  categoryName,
+  dispatch
+}: {
+  categoryKey: string;
+  categoryName: string;
+  active: boolean;
+  dispatch: React.Dispatch<ReminderSettingsAction>;
+}) => {
+  const [switchValue, setSwitchValue] = useState(active);
+
+  const toggleSwitch = useCallback(
+    (value) => {
+      setSwitchValue(value);
+      dispatch({
+        type: ReminderSettingsActionType.UPDATE_ACTIVE_TYPE,
+        payload: { key: categoryKey, value }
+      });
+    },
+    [categoryKey, dispatch, setSwitchValue]
+  );
+
+  const onPress = useCallback(() => {
+    toggleSwitch(!switchValue);
+  }, [switchValue, toggleSwitch]);
+
+  useEffect(() => {
+    setSwitchValue(active);
+  }, [active, setSwitchValue]);
+
+  return (
+    <ListItem
+      title={<RegularText>{categoryName}</RegularText>}
+      bottomDivider
+      rightIcon={<Switch switchValue={switchValue ?? false} toggleSwitch={toggleSwitch} />}
```

Comment:
on android the switch is not animating here. didn't we fix that in our call?

Reply:
we did....
and we did not...
we set the component to be the touchable, which caused it to be animated.
it seems that changing the component fixes it with fast refresh. so i have no idea how to go about it at the moment.

Reply:
ok, strange but fine